lib/jazzy/doc_builder.rb

```diff
@@ -235,6 +238,45 @@ def self.document_markdown(source_module, doc_model, path_to_root)
       doc.render
     end
 
+    # Returns the appropriate color for the provided percentage,
+    # used for generating a badge on shields.io
+    # @param [Number] coverage The documentation percentage
+    def self.color_for_coverage(coverage)
+      if coverage < 10
+        return 'red'
+      elsif coverage < 30
+        return 'orange'
+      elsif coverage < 60
+        return 'yellow'
+      elsif coverage < 85
+        return 'yellowgreen'
+      elsif coverage < 90
+        return 'green'
+      else
+        return 'brightgreen'
+      end
+    end
+
+    # Downloads an SVG from shields.io displaying the documentation percentage
+    # @param [SourceModule] source_module The source module from SourceKitten
+    # @param [Config] options Build options
+    def self.download_badge(source_module, options)
```

Comment:
The only thing we need from `source_module` in here is `doc_coverage`, so why not make this method a bit more modular and composable by just taking that as an argument? e.g. `download_badge(coverage, options)`